lib/dev/kit/system.rb

```diff
@@ -44,8 +45,39 @@ def sudo_reason(msg)
         # #### Usage
         # `out, stat = Dev::Kit::System.capture2('ls', 'a_folder')`
         #
-        def capture2(*a, sudo: false, env: ENV, **kwargs)
-          delegate_open3(*a, sudo: sudo, env: env, method: :capture2, **kwargs)
+        def capture2(*a, sudo: false, env: ENV, stdin_data: '')
+          write_thread = nil
+          a = apply_sudo(*a, sudo)
+
+          in_r, in_w = IO.pipe
+          out_r, out_w = IO.pipe
+          err_r, err_w = IO.pipe
+          pid = Process.spawn(env, *resolve_path(a, env), :in => in_r, :out => out_w, :err => err_w)
+          out_w.close
+          err_w.close
+          in_w.write(stdin_data)
+          in_w.close
+          in_r.close
+          write_thread = Thread.new do
```

Comment:
Using select, I would need to use non blocking writes or something. Because in the deadlock situation you described we would be dead locked on the call to write